### PR TITLE
Split the Routes file. Closes #40

### LINKED ISSUE #40: Split the Routes file
Split the Routes file in many files

### PR DESCRIPTION
**Pull request by danpersa**

### CODE REVIEW

File: src/main/scala/org/zalando/spearheads/innkeeper/Routes.scala [hunk 1]
```diff
@@ -1,46 +1,33 @@
...
-class Routes @Inject() (implicit val materializer: ActorMaterializer,
-                        val routesService: RoutesService,
-                        val teamService: TeamService,
-                        val jsonService: JsonService,
-                        val scopes: Scopes,
+class Routes @Inject() (val getRoute: GetRoute,
+                        val getRoutes: GetRoutes,
+                        val deleteRoute: DeleteRoute,
+                        val postRoutes: PostRoutes,
+                        val getUpdatedRoutes: GetUpdatedRoutes,
```
Comment (mo-gr) [inline src/main/scala/org/zalando/spearheads/innkeeper/Routes.scala]: Interesting solution of injecting all the actual routes. I think I like it.

File: src/main/scala/org/zalando/spearheads/innkeeper/routes/DeleteRoute.scala [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        }
+      }
+    }
+  }
+
+  private def deleteRoute(id: Long) = {
+    onComplete(routesService.remove(id)) {
+      case Success(ServiceResult.Success(_))        => complete("")
+      case Success(ServiceResult.Failure(NotFound)) => complete(StatusCodes.NotFound)
+      case Success(_)                               => complete(StatusCodes.NotFound)
```
Comment (mo-gr) [inline src/main/scala/org/zalando/spearheads/innkeeper/routes/DeleteRoute.scala]: this line looks curious... Success -> NotFound? Is that correct?

File: src/main/scala/org/zalando/spearheads/innkeeper/routes/PostRoutes.scala [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+  private val LOG = LoggerFactory.getLogger(this.getClass)
+
+  def apply(authenticatedUser: AuthenticatedUser, token: String): Route = {
+    post {
+      LOG.info("post /routes/")
+      entity(as[RouteIn]) { route =>
+        LOG.debug(s"route ${route}")
+        team(authenticatedUser, token)(teamService) { team =>
+          (isRegexRoute(route.route) & hasOneOfTheScopes(authenticatedUser)(scopes.WRITE_REGEX)) {
```
Comment (mo-gr) [inline src/main/scala/org/zalando/spearheads/innkeeper/routes/PostRoutes.scala]: greedy and? Shouldn't the `&` be `&&` ? Currently it always eval hasOneOfTheScopes even if isRegexRoute is already false.

Comment (danpersa) [inline src/main/scala/org/zalando/spearheads/innkeeper/routes/PostRoutes.scala]: `&` is part of the akka http DSL... not related to &&...

akka http uses `&` and `|`

interesting, isn't it?

File: src/main/scala/org/zalando/spearheads/innkeeper/routes/DeleteRoute.scala (hunk 2 above)
Comment (danpersa) [inline src/main/scala/org/zalando/spearheads/innkeeper/routes/DeleteRoute.scala]: yes, it's correct, as this case shouldn't happen

**Pull request merged**